web/__init__.py

```diff
@@ -98,6 +129,21 @@ def app_api_version():
     }
 
 
+@app.post('/api/ankerctl/config/upload')
+def app_api_ankerctl_config_upload():
+    with tempfile.TemporaryDirectory(prefix='ankerctl_') as tmpdir:
+        if request.method != 'POST':
+            return web.util.flash_redirect()
+        if 'login_file' not in request.files:
+            return web.util.flash_redirect('No file found', 'danger')
+        file = request.files['login_file']
+        if file.filename == '':
+            return web.util.flash_redirect('No file uploaded', 'danger')
```

Comment:
It's absolutely possible to upload a file without specifying a file name (quite easy mistake to make with `curl`, for example).

There's no reason to reject the upload, if the file is valid. We should just run `web.config.config_import` on it.

Reply:
This `web.config.config_import` should raise an exception if something goes wrong, and then we catch it here.

`flash_redirect()` shouldn't find its way into every corner of the codebase - ideally, we should only use it in the direct request handler functions.